packages/block-editor/src/components/inserter/index.js

```diff
@@ -105,10 +122,72 @@ class Inserter extends Component {
 
 export default compose( [
 	withSelect( ( select, { rootClientId } ) => {
-		const { hasInserterItems } = select( 'core/block-editor' );
+		const {
+			hasInserterItems,
+			__experimentalGetAllowedBlocks,
+		} = select( 'core/block-editor' );
+
+		const allowedBlocks = __experimentalGetAllowedBlocks( rootClientId );
+
+		const hasOnlyOneAllowedInserterBlockType = allowedBlocks && ( get( allowedBlocks, [ 'length' ], 0 ) === 1 );
```

Comment:
this is a bug introduced by all the refactoring, will check!

Reply:
It wasn't a but in my code, but the default button appender was simply printing __('Add blocl') instead of the label I have supplied. I have also added in this PR a tooltip for the default button appender.